src/tbb/rml_thread_monitor.h

```diff
@@ -183,6 +184,33 @@ inline void thread_monitor::check( int error_code, const char* routine ) {
     }
 }
 
+// pthread_create(2) can spuriously fail on Linux. This is a function
+// to wrap pthread_create(2) to retry if it fails with EAGAIN.
+inline void do_pthread_create(pthread_t *handle, pthread_attr_t *s, void* (*thread_routine)(void*), void* arg) {
+#ifdef __linux__
+  const int max_num_tries = 20;
+  int i = 0;
+
+  for (;;) {
+    int error_code = pthread_create(handle, s, thread_routine, arg);
+    if (error_code == 0)
+      return;
+
+    if (error_code != EAGAIN || i++ > max_num_tries) {
```

Comment:
This expression a bit complicated, that why I suggested move increment to another place.

Reply:
Done